src/VisualStudio/Core/Impl/SolutionExplorer/AnalyzerItem/AnalyzerItemSource.cs

```diff
@@ -2,237 +2,174 @@
-using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
 using Microsoft.VisualStudio.Shell;
+using Roslyn.Utilities;
 
-namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplorer
-{
-    internal class AnalyzerItemSource : IAttachedCollectionSource, INotifyPropertyChanged
-    {
-        private readonly AnalyzersFolderItem _analyzersFolder;
-        private readonly IAnalyzersCommandHandler _commandHandler;
-        private IReadOnlyCollection<AnalyzerReference> _analyzerReferences;
-        private BulkObservableCollection<AnalyzerItem> _analyzerItems;
-
-        public event PropertyChangedEventHandler PropertyChanged;
-
-        public AnalyzerItemSource(AnalyzersFolderItem analyzersFolder, IAnalyzersCommandHandler commandHandler)
-        {
-            _analyzersFolder = analyzersFolder;
-            _commandHandler = commandHandler;
-
-            _analyzersFolder.Workspace.WorkspaceChanged += Workspace_WorkspaceChanged;
-        }
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplorer;
 
-        private void Workspace_WorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
-        {
-            switch (e.Kind)
-            {
-                case WorkspaceChangeKind.SolutionAdded:
-                case WorkspaceChangeKind.SolutionChanged:
-                case WorkspaceChangeKind.SolutionReloaded:
-                    UpdateAnalyzers();
-                    break;
-
-                case WorkspaceChangeKind.SolutionRemoved:
-                case WorkspaceChangeKind.SolutionCleared:
-                    _analyzersFolder.Workspace.WorkspaceChanged -= Workspace_WorkspaceChanged;
-                    break;
-
-                case WorkspaceChangeKind.ProjectAdded:
-                case WorkspaceChangeKind.ProjectReloaded:
-                case WorkspaceChangeKind.ProjectChanged:
-                    if (e.ProjectId == _analyzersFolder.ProjectId)
-                    {
-                        UpdateAnalyzers();
-                    }
-
-                    break;
-
-                case WorkspaceChangeKind.ProjectRemoved:
-                    if (e.ProjectId == _analyzersFolder.ProjectId)
-                    {
-                        _analyzersFolder.Workspace.WorkspaceChanged -= Workspace_WorkspaceChanged;
-                    }
-
-                    break;
-            }
-        }
+internal sealed class AnalyzerItemSource : IAttachedCollectionSource, INotifyPropertyChanged
+{
+    private readonly AnalyzersFolderItem _analyzersFolder;
+    private readonly IAnalyzersCommandHandler _commandHandler;
 
-        private void UpdateAnalyzers()
-        {
-            if (_analyzerItems == null)
-            {
-                // The set of AnalyzerItems hasn't been realized yet. Just signal that HasItems
-                // may have changed.
+    private readonly BulkObservableCollection<AnalyzerItem> _items = [];
```

Comment:
instead of a collection we overwrite, we now just use a BOC as that can be mutated in place while handling notification to appropriate listeners.